tests/test_data.py

```diff
@@ -4,213 +4,155 @@
-        "returns_srcid": "S2A_MSIL2A_20200221T102041_N0214_R065_T32UQC_20200221T120618",
-        "returns_uuid": "560f78fb-22b8-4904-87de-160d9236d33e",
-    },
-    {
-        "datahub": Datahub.Scihub,
-        "catalog": None,
-        "platform_name": Platform.Sentinel3,
-        "date": ("20200220", "20200225"),
-        "aoi": aoi_bbox,
-        "cloud_cover": (0, 100),
-        "returns_srcid": "S3B_OL_2_LRR____20200220T092808_20200220T101154_20200221T143235_2626_035_364______LN1_O_NT_002",
-        "returns_uuid": "a50c1e2f-0688-4be1-ae2f-dc69fe8b170a",
-    },
-    {
-        "datahub": Datahub.EarthExplorer,
-        "catalog": None,
-        "platform_name": Platform.Landsat5,
-        "date": ("20100201", "20100225"),
-        "aoi": aoi_4326,
-        "cloud_cover": (0, 100),
-        "returns_srcid": "LT05_L1GS_193024_20100207_20161016_01_T2",
-        "returns_uuid": "LT51930242010038MOR00",
-    },
-    {
-        "datahub": Datahub.EarthExplorer,
-        "catalog": None,
-        "platform_name": Platform.Landsat7,
-        "date": ("20150810", "20150825"),
-        "aoi": aoi_3857,
-        "cloud_cover": (0, 50),
-        "returns_srcid": "LE07_L1TP_193024_20150824_20161025_01_T1",
-        "returns_uuid": "LE71930242015236ASN00",
-    },
-    {
-        "datahub": Datahub.EarthExplorer,
-        "catalog": None,
-        "platform_name": Platform.Landsat8,
-        "date": ("20200310", "20200325"),
-        "aoi": aoi_bbox,
-        "cloud_cover": (0, 20),
-        "returns_srcid": "LC08_L1TP_193024_20200322_20200326_01_T1",
-        "returns_uuid": "LC81930242020082LGN00",
-    },
-    {
-        "datahub": Datahub.STAC_local,
-        "catalog": catalog_path,
-        "platform_name": Platform.Sentinel2,
-        "date": ("20200220", "20200225"),
-        "aoi": aoi_3857,
-        "cloud_cover": (90, 100),
-        "returns_srcid": "S2A_MSIL1C_20200221T102041_N0209_R065_T32UPC_20200221T110731",
-        "returns_uuid": "ae674e64-013d-4898-a6d7-096d7b02bdde",
-    },
-]
-
 
 class DataTest(unittest.TestCase):
-    def setUp(self):
-        with Source(datahub=queries[2]["datahub"], catalog=queries[2]["catalog"]) as src:
-            meta = src.query_metadata(
-                platform=queries[2]["platform_name"],
-                date=queries[2]["date"],
-                aoi=queries[2]["aoi"],
-                cloud_cover=queries[2]["cloud_cover"],
-            )
-            meta.normalize_hrefs(Path(gettempdir()).as_posix())
-            meta.save()  # used for testing Datahub.file
-
-    def tearDown(self):
-        catalog_path.unlink()
-
-    def test_init(self):
+    def test_init_stac_catalog(self):
         with Source(datahub=Datahub.STAC_local, catalog=catalog_path) as src:
             self.assertTrue(isinstance(src.api, pystac.catalog.Catalog))
 
+    def test_init_stac_url(self):
+        with Source(datahub=Datahub.STAC_API, url=r"https://earth-search.aws.element84.com/v0/") as src:

```

Comment:
Aren't we again dependent on the availability of an external datahub here? Perhaps there is a way to also mock this?

Reply:
Sure, we could do this. This was just way more reliable so far.

Reply:
Also we are not using another library for the actual connection here, so I would suggest we keep it until it makes problems.